ocs_ci/ocs/longevity_stage2.py

```diff
@@ -0,0 +1,561 @@
+                1, bucketclass=source_bucketclass, replication_policy=replication_policy
+            )[0]
+            source_buckets.append(source_bucket)
+
+            write_empty_files_to_bucket(
+                mcg_obj, awscli_pod_session, source_bucket.name, test_directory_setup
+            )
+            compare_bucket_object_list(
+                mcg_obj_session, source_bucket.name, target_bucket_name
+            )
+        obc_objs.extend(target_buckets)
+        obc_objs.extend(source_buckets)
+
+        for obc in obc_objs:
+            obc_names.append(obc.name)
+
+        if measure:
+            # Measure OBC Creation Time
+            scale_noobaa_lib.measure_obc_creation_time(obc_name_list=obc_names)
+
+        # Delete OBCs
+        for bucket in obc_objs:
+            logger.info(f"Deleting bucket: {bucket.name}")
+            bucket.delete()
+
+        if measure:
+            # Measure OBC Deletion Time
+            scale_noobaa_lib.measure_obc_deletion_time(obc_name_list=obc_names)
+
+    return factory
+
+
+def stage2(
+    multi_pvc_pod_lifecycle_factory,
+    multi_obc_lifecycle_factory,
+    num_of_pvcs=100,
+    pvc_size=2,
+    num_of_obcs=20,
+    run_time=1440,
+    measure=False,
+):
+    """
+    Function to handle automation of Longevity Stage 2 Sequential Steps i.e. Creation / Deletion of PVCs, PODs and OBCs
+    and measurement of creation / deletion times of the mentioned resources.
+
+    Args:
+        multi_pvc_pod_lifecycle_factory : Fixture to create/delete multiple pvcs and pods and
+                                            measure pvc creation/deletion time and pod attach time.
+        multi_obc_lifecycle_factory : Fixture to create/delete multiple obcs and
+                                        measure their creation/deletion time.
+        num_of_pvcs (int) : Total Number of PVCs / PODs we want to create.
+        pvc_size (int) : Size of each PVC in GB.
+        num_of_obcs (int) : Number of OBCs we want to create of each type. (Total OBCs = num_of_obcs * 5)
+        run_time (int) : Total Run Time in minutes.
+        measure (bool) : True if we want to measure the performance metrics, False otherwise.
+
+    """
+    end_time = datetime.now() + timedelta(minutes=run_time)
+    cycle_no = 0
+
+    while datetime.now() < end_time:
+        cycle_no += 1
+        logger.info(f"------------STARTING CYCLE:{cycle_no}------------")
+
+        for bulk in (False, True):
+            logger.info(f"-----BULK:{bulk}------")
+            multi_pvc_pod_lifecycle_factory(
+                num_of_pvcs=num_of_pvcs,
+                pvc_size=pvc_size,
+                bulk=bulk,
+                namespace="stage-2-cycle-" + str(cycle_no),
+                measure=measure,
+            )
+            multi_obc_lifecycle_factory(
+                num_of_obcs=num_of_obcs, bulk=bulk, measure=measure
+            )
+
+        logger.info(f"------------ENDING CYCLE:{cycle_no}------------")
```

Comment:
It will be good if we have some delay between sequential and bulk creation and also between the cycles to mimic to some extent the real world scenarios.

Reply:
Done